pkg/manager/manager.go

```diff
@@ -311,8 +311,9 @@ func (m *Manager) handleDaemonDeathEvent() {
 			log.L.Warnf("Daemon %s was not found", ev.daemonID)
 			return
 		}
-		collector.NewDaemonInfoCollector(&d.Version, -1).Collect()
+
 		d.Lock()
+		collector.NewDaemonInfoCollector(&d.Version, -1).Collect()
```

Comment:
Whether is a lock needed before another call to this method?

Reply:
In a few cases, I get the extra values that `d.version` is incorrect. However, the value of the expected version is eventually correct.

Reply:
What do you mean by "incorrect"?  The version string is shorter than expected or polluted?

Reply:
The `version` string is null. It appeared by chance.